x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MlInitializationService.java

```diff
@@ -94,9 +95,18 @@ public void clusterChanged(ClusterChangedEvent event) {
             return;
         }
 
+        final boolean isCurrentResetMode = MlMetadata.getMlMetadata(event.state()).isResetMode();
+        final boolean isPreviousResetMode = MlMetadata.getMlMetadata(event.previousState()).isResetMode();
+
         // The atomic flag prevents multiple simultaneous attempts to create the
         // index if there is a flurry of cluster state updates in quick succession
-        if (this.isMaster && isIndexCreationInProgress.compareAndSet(false, true)) {
+        if (this.isMaster
+            && isIndexCreationInProgress.compareAndSet(false, true)
+            // Don't bother checking if the current state is in reset mode
+            && isCurrentResetMode == false
+            // If the previous state was reset mode, it means we recently changed it to off, so don't immediate create annotations index
+            && isPreviousResetMode == false
```

Comment:
If CI is slow enough, I ran into this multiple times where `reset_mode` is unset, MlInitizliationService starts creating annotation index, then immediately something tries to cleanup annotations again (in test cleanup). 

I think not firing when the `reset_mode` is unset in the current cluster update is acceptable.

Reply:
To me this implies there's a problem with feature reset.  Feature reset should not return until all the indices it was supposed to delete have been fully deleted.

I have created #76157 to see if I can reproduce the errors you were seeing.